modules/tracktion_engine/playback/graph/tracktion_MidiInputDeviceNode.cpp

```diff
@@ -202,4 +202,9 @@ bool MidiInputDeviceNode::isLivePlayOverActive()
     return instance.isRecording() && instance.context.transport.looping;
 }
 
+void MidiInputDeviceNode::discardRecordings()
+{
+    liveRecordedMessages.clear();
```

Comment:
It won't be thread safe to call this as the audio thread could still be writing to the array.
Guard it with a `const juce::ScopedLock sl (liveInputLock);` like the other uses in the class.